buildlib/tools/builds.sh

```diff
@@ -461,5 +461,5 @@ do
 	[ -d "${ucx_build_dir}" ] && rm -rf ${ucx_build_dir}/*
 
 	# run the test
-	$test_name || { azure_log_error "Test failed: $test_name"; exit 1; }
+	$test_name
```

Comment:
how did it hid the build issues?

Reply:
when using `||` or `if !` on a bash function, seems `set -e` is ignored:
```shell
#!/bin/bash -eEl

run() {
    echo options set $-
    # grep returns error $? 2 like make
    grep abc abc
    echo Should never be seen
}

echo == broken 1
run || echo FAIL 1

echo == broken 2
if ! run; then
    echo FAIL 2
fi

echo == broken 3
( run ) || echo FAIL 3

echo == broken 3
{ run; } || echo FAIL 4

echo == success
run
echo Should not be seen 2
```

```
$ ./a.sh
== broken 1
options set ehBE
grep: abc: No such file or directory
Should never be seen
== broken 2
options set ehBE
grep: abc: No such file or directory
Should never be seen
== broken 3
options set ehBE
grep: abc: No such file or directory
Should never be seen
== broken 3
options set ehBE
grep: abc: No such file or directory
Should never be seen
== success
options set ehBE
grep: abc: No such file or directory
```